src/back/tests/e2e/users.test.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file Test file for user related end-points.
+ */
+import { expect } from 'chai';
+
+import testLib from '../lib/test.lib';
+import gqlClient from '../lib/gqlClient.lib';
+import { queryAllUsers, querySingleUser } from '../queries/users';
+
+// init:
+testLib();
+
+describe('User end-points', () => {
+  describe('Query All Users', () => {
+    it('returns a list of users with each of their attributes', async () => {
+      const { data: { data: { users } } } = await gqlClient.query(queryAllUsers);
+      expect(users.length).to.equal(25);
+      users.forEach((user) => {
+        expect(user).to.haveOwnPropertyDescriptor('id');
+        expect(user).to.haveOwnPropertyDescriptor('name');
+        expect(user).to.haveOwnPropertyDescriptor('username');
+        expect(user).to.haveOwnPropertyDescriptor('email');
+      });
+    });
+
+    it('returns a sliced list when asked for it', async () => {
+      const { data: { data: { users } } } = await gqlClient.query(queryAllUsers, {
+        variables: {
+          slice: { limit: 2, offset: 4 },
+        },
+      });
+
+      expect(users.length).to.equal(2);
+      expect(users[0].id).to.equal(5);
```

Comment:
Since the **username** is a unique factor, I think it'd be better to use it instead of the ID. Still, assuming we're not intending to hard-delete any users, this might work as well. Still, only taking the fact into consideration.

Reply:
For the tests, this is not relevant. Tests use a newly generated DB every time, and the DB we generate goes always from 1-100, so I can be 100% sure this test should pass.